### PR TITLE
UICAL-272: Added TitleManager component to views to add html page titles

### PR DESCRIPTION
**Pull request by danetsao**
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added an appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally, all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/views/CreateEditCalendarLayer.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ import {
   PaneFooter,
   Paneset,
 } from '@folio/stripes/components';
+import { TitleManager } from '@folio/stripes/core/index.js';
```
Comment (ncovercash) [inline src/views/CreateEditCalendarLayer.tsx]: Should just be `@folio/stripes/core`

File: src/views/AllCalendarView.tsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import {
   Pane
 } from '@folio/stripes/components';
 import { IfPermission, useStripes } from '@folio/stripes/core';
+import { TitleManager } from '@folio/stripes/core/index.js';
```
Comment (ncovercash) [inline src/views/AllCalendarView.tsx]: Should just be `@folio/stripes/core` (a few other places too)

File: src/views/CurrentAssignmentView.tsx [hunk 3]
```diff
@@ -34,7 +35,7 @@ export const CurrentAssignmentView: FunctionComponent<
   const history = useHistory();
   const currentRouteId = useRouteMatch<{ servicePointId: string }>(
     '/settings/calendar/active/:servicePointId'
-  )?.params?.servicePointId;
+  )?.params?.servicePointId || '';
```
Comment (ncovercash) [inline src/views/CurrentAssignmentView.tsx]: `??` is preferred over `||`

File: src/views/CurrentAssignmentView.tsx [hunk 4]
```diff
@@ -92,8 +93,14 @@ export const CurrentAssignmentView: FunctionComponent<
     };
   });
 
+  const calendarName = dataRepository.getCalendars().filter((c) => c.assignments.includes(currentRouteId))[0]?.name || '';
```
Comment (ncovercash) [inline src/views/CurrentAssignmentView.tsx]: `??` is preferred over `||`

**Review by ncovercash — CHANGES_REQUESTED** [review]
Fix imports and tests and you should be good!

File: src/typings/stripes/core.d.ts [hunk 5]
```diff
@@ -21,6 +25,14 @@ export const IfPermission: FunctionComponent<{
   children: ReactNode | ((props: { hasPermission: boolean }) => ReactNode);
 }>;
 
+export const TitleManager: React.Component<{
```
Comment (ncovercash) [inline src/typings/stripes/core.d.ts]:
```suggestion
export const TitleManager: FunctionComponent<{
```

Should specify `ComponentType` or `FunctionComponent` — `Component` alone is something else unfortunately

Comment (danetsao) [reply src/typings/stripes/core.d.ts]: Ok, that fixed local errors as well. I changed to FunctionComponent.

**Review by ncovercash — APPROVED** [review]

**Review by nhanaa — APPROVED** [review]

**Pull request merged**